sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3918,6 +3919,64 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         c.storeAnalyzedQuery()
     }
   }
+
+  object ResolveTranspose extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsWithPruning(
+      _.containsPattern(TRANSPOSE), ruleId) {
+      case t @ Transpose(columnsToTranspose, columnAlias, orderExpression, child) =>
+//      case t @ Transpose(columnsToTranspose, columnAlias, orderExpression, child)
+//        if t.expressions.forall(_.resolved) && child.resolved =>
+
+        // scalastyle:off println
+        val pivotColumnName = "__pivot_column"
+
+        val concatExpr = ConcatWs(Literal("_") +: columnsToTranspose)
+        val aliasExpr = Alias(concatExpr, pivotColumnName)()
+        val tempChild = Project(child.output :+ aliasExpr, child)
+
+        val columnsToKeep = child.output.filterNot(columnsToTranspose.contains)
+
+        val pivotValues = columnsToTranspose.map(_.asInstanceOf[NamedExpression])
+
+
+        println("Unpivot ids: " + Some(Seq(AttributeReference(pivotColumnName, StringType)())))
```

Comment:
To debug: Unpivot has empty schema
```
Unpivot ids: Some(List(__pivot_column#15))
Unpivot values: Some(List(List(year#5), List(earnings#6)))
Unpivot variableColumnName: variable
Unpivot valueColumnNames: List(value)
Unpivot child: root
 |-- course: string (nullable = true)
 |-- year: integer (nullable = false)
 |-- earnings: double (nullable = false)
 |-- __pivot_column: string (nullable = false)
Unpivot logical node: 'Unpivot List(__pivot_column#16), List(List(year#5), List(earnings#6)), variable, [value]
+- Project [course#4, year#5, earnings#6, concat_ws(_, course#4) AS __pivot_column#14]
   +- SerializeFromObject [staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$CourseSales, true])).course, true, false, true) AS course#4, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$CourseSales, true])).year AS year#5, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$CourseSales, true])).earnings AS earnings#6]
      +- ExternalRDD [obj#3]
Unpivot schema: root
```

Reply:
And because of the empty Unpivot, 
```
[info] - logical plan transformation for transpose *** FAILED *** (1 second, 133 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: Aggregate expression required for pivot, but 'value' did not appear in any aggregate function.
```
is raised in Pivot